src/eddymotion/model.py

```diff
@@ -271,6 +270,102 @@ def predict(self, gradient, **kwargs):
         return self._data
 
 
+class PETModel:
+    """A PET imaging realignment model based on B-Spline approximation."""
+
+    __slots__ = ("_t", "_x", "_order", "_coeff", "_mask", "_shape", "_n_ctrl")
+
+    def __init__(self, timepoints=None, n_ctrl=None, mask=None, order=3, **kwargs):
+        """
+        Create the B-Spline interpolating matrix.
+
+        Parameters:
+        -----------
+        timepoints : :obj:`list`
+            The timing (in sec) of each PET volume.
+            E.g., ``[20.,   40.,   60.,  120.,  180.,  240.,  360.,  480.,  600.,
```

Comment:
Is it common for PET to be acquired with this kind of non-uniform temporal sampling? (I don't know much about PET, so forgive the potentially naive question)

Reply:
I haven't checked the BIDS specs, but it is my impression that you can define frame starts or frame durations, which are typically nonregular (@mnoergaard is in a much better position to weigh in)

However, your comment has made me realize that the model should be fit at the frame midpoints instead of the starts.

Reply:
Yes - I would say this is quite common (for dynamics PET with a bolus injection), but there are also cases where uniform sampling is used (infusion studies).